src/Controller/SubmissionController.php

```diff
@@ -303,4 +308,82 @@ protected function hasRightsToViewForum($forum) {
         }
         return false;
     }
+
+    public function importFromReddit(EntityManager $em, Request $request, SubmissionRepository $sr, UserRepository $ur, ForumRepository $fr, Translator $tr = null) {
+        // LOCAL ONLY BRUH
+        if ($request->getClientIp() != "127.0.0.1") return;
+
+        // Prime Response
+        $successReport = [
+          "success" => 'false',
+          "imported" => array(),
+          "ignored" => array()
+        ];
+
+        // Check top 10 every time.
+        $redditUrl = "https://www.reddit.com/r/gundeals/new.json?sort=new&limit=10";
+
+        try {
+          $ch = curl_init();
+          curl_setopt($ch, CURLOPT_URL, $redditUrl);
+          curl_setopt($ch, CURLOPT_HTTPHEADER, array('Content-type: application/json')); // Assuming you're requesting JSON
+          curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+
+          $response = curl_exec($ch);
+
+          // The goods.
+          $data = json_decode($response);
+          foreach ($data->data->children as $posting)
+          {
+            $url = $posting->data->url;
+            $title = $posting->data->title;
+            $permalink = $posting->data->permalink;
+            $hidden = $posting->data->hidden;
+            $pinned = $posting->data->pinned;
+            $modbanned = $posting->data->banned_by;
+
+            // Only import non banned and non hidden and non pinned posts
+            if ($hidden == false && $modbanned == null && $pinned == false){
+              // We got that url? YAAA BOY!
+              if ($url != null && $title != null)
+              {
+
+                /* Check for submission with url of incoming url */
+                $foundEntry = $sr->findOneByUrl($url);
+                if ($foundEntry == null){
+                  /* If not, create submission */
+                  $forum = $fr->findOneByCaseInsensitiveName("gundeals");
+
+                  $data = new SubmissionData($forum);
+                    $data->setTitle($title);
+                    $data->setUrl($url);
+                    $data->setBody("Imported From /r/gundeals. | " & $permalink);
+                    $data->setSticky(false);
+                    $data->setModThread(false);
+
+                  $user = $ur->loadUserByUsername("gundeals");
```

Comment:
Both `$user` and `$forum` could be loaded outside of the for loop, so as to not pull that information for every single post.